### ISSUE TITLE
Evaluation script `lm_eval_harness.py` failing

### ISSUE THREAD

**Issue by akjindal53244** [Lightning-AI/lit-gpt#442]
I am following instructions [NeurIPS 2023 LLM Efficiency Challenge Quickstart Guide](https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/neurips_challenge_quickstart.md) here and local evaluation script fails.

Command: `python eval/lm_eval_harness.py --checkpoint_dir checkpoints/stabilityai/stablelm-base-alpha-3b --precision "bf16-true" --eval_tasks "[gsm8k]" --batch_size 4 --save_filepath "results-stablelm-3b_gsm8k.json"`

```
Running greedy_until requests
  0%|                                                                                                                                      | 0/1319 [00:03<?, ?it/s]
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/minimalist/work/projects/lit-gpt/eval/lm_eval_harness.py:254 in <module>                   │
│                                                                                                  │
│   251 │   │   "ignore",                                                                          │
│   252 │   │   message="ComplexHalf support is experimental and many operators don't support it   │
│   253 │   )                                                                                      │
│ ❱ 254 │   result = CLI(run_eval_harness)                                                         │
│   255 │   print(result)                                                                          │
│   256                                                                                            │
│                                                                                                  │
│ /home/minimalist/miniconda3/envs/nips/lib/python3.9/site-packages/jsonargparse/_cli.py:85 in CLI │
│                                                                                                  │
│    82 │   │   │   return parser                                                                  │
│    83 │   │   cfg = parser.parse_args(args)                                                      │
│    84 │   │   cfg_init = parser.instantiate_classes(cfg)                                         │
│ ❱  85 │   │   return _run_component(component, cfg_init)                                         │
│    86 │                                                                                          │
│    87 │   subcommands = parser.add_subcommands(required=True)                                    │
│    88 │   comp_dict = {c.__name__: c for c in components}                                        │
│                                                                                                  │
│ /home/minimalist/miniconda3/envs/nips/lib/python3.9/site-packages/jsonargparse/_cli.py:147 in    │
│ _run_component                                                                                   │
│                                                                                                  │
│   144 def _run_component(component, cfg):                                                        │
│   145 │   cfg.pop("config", None)                                                                │
│   146 │   if not inspect.isclass(component):                                                     │
│ ❱ 147 │   │   return component(**cfg)                                                            │
│   148 │   subcommand = cfg.pop("subcommand")                                                     │
│   149 │   if not subcommand:                                                                     │
│   150 │   │   return component(**cfg)                                                            │
│                                                                                                  │
│ /home/minimalist/work/projects/lit-gpt/eval/lm_eval_harness.py:234 in run_eval_harness           │
│                                                                                                  │
│   231 │   │   quantize=quantize,                                                                 │
│   232 │   )                                                                                      │
│   233 │   eval_harness.fabric.print("Running evaluation harness...")                             │
│ ❱ 234 │   results = eval_harness.run_eval(                                                       │
│   235 │   │   eval_tasks=eval_tasks, num_fewshot=num_fewshot, bootstrap_iters=bootstrap_iters,   │
│   236 │   )                                                                                      │
│   237 │   if save_filepath:                                                                      │
│                                                                                                  │
│ /home/minimalist/miniconda3/envs/nips/lib/python3.9/site-packages/torch/utils/_contextlib.py:115 │
│ in decorate_context                                                                              │
│                                                                                                  │
│   112 │   @functools.wraps(func)                                                                 │
│   113 │   def decorate_context(*args, **kwargs):                                                 │
│   114 │   │   with ctx_factory():                                                                │
│ ❱ 115 │   │   │   return func(*args, **kwargs)                                                   │
│   116 │                                                                                          │
│   117 │   return decorate_context                                                                │
│   118                                                                                            │
│                                                                                                  │
│ /home/minimalist/work/projects/lit-gpt/eval/lm_eval_harness.py:184 in run_eval                   │
│                                                                                                  │
│   181 │   │   │   lm = base.CachingLM(lm, "lm_cache/" + name + ".db")                            │
│   182 │   │                                                                                      │
│   183 │   │   print(f"lm: {lm}, task_dict: {tasks.get_task_dict(eval_tasks)}, num_fewshot: {nu   │
│ ❱ 184 │   │   results = evaluator.evaluate(                                                      │
│   185 │   │   │   lm=lm,                                                                         │
│   186 │   │   │   │   task_dict=tasks.get_task_dict(eval_tasks),                                 │
│   187 │   │   │   description_dict=description_dict,                                             │
│                                                                                                  │
│ /home/minimalist/work/projects/lm-evaluation-harness/lm_eval/utils.py:243 in _wrapper            │
│                                                                                                  │
│   240 │   │   │   │   "deprecated and will be disallowed in a future version of "                │
│   241 │   │   │   │   "lm-evaluation-harness!"                                                   │
│   242 │   │   │   )                                                                              │
│ ❱ 243 │   │   return fn(*args, **kwargs)                                                         │
│   244 │                                                                                          │
│   245 │   return _wrapper                                                                        │
│   246                                                                                            │
│                                                                                                  │
│ /home/minimalist/work/projects/lm-evaluation-harness/lm_eval/evaluator.py:305 in evaluate        │
│                                                                                                  │
│   302 │   │   #       they should end up next to each other.                                     │
│   303 │   │                                                                                      │
│   304 │   │   print("Running", reqtype, "requests")                                              │
│ ❱ 305 │   │   resps = getattr(lm, reqtype)([req.args for req in reqs])                           │
│   306 │   │   resps = [                                                                          │
│   307 │   │   │   x if req.index is None else x[req.index] for x, req in zip(resps, reqs)        │
│   308 │   │   ]                                                                                  │
│                                                                                                  │
│ /home/minimalist/work/projects/lm-evaluation-harness/lm_eval/base.py:457 in greedy_until         │
│                                                                                                  │
│   454 │   │   │   │   context_enc, context_enc.shape[1] + max_gen_tokens, primary_until          │
│   455 │   │   │   )                                                                              │
│   456 │   │   │                                                                                  │
│ ❱ 457 │   │   │   s = self.tok_decode(cont[0].tolist()[context_enc.shape[1] :])                  │
│   458 │   │   │                                                                                  │
│   459 │   │   │   for term in until:                                                             │
│   460 │   │   │   │   s = s.split(term)[0]                                                       │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'str' object has no attribute 'tolist'

```

**Comment by timothylimyl** [comment]
lm_eval for lora is buggy too. It cannot import scripts.prepare_alpaca

lm_eval for normal has bug at the end of eval where it says:
```
TypeError: generate() got an unexpected keyword argument 'max_new_tokens'
```

**Comment by timothylimyl** [comment]
potential fix over at lm-eval:

```
            s = self._model_generate(
                context_enc, context_enc.shape[1] + max_gen_tokens, primary_until
            )
            # s = self.tok_decode(cont[0].tolist()[context_enc.shape[1] :])
```

comment out the decoding since it is already done.

**Comment by carmocca** [comment]
Probably fixed by #752
